### PR TITLE
bank-account: add to track

### PR DESCRIPTION
**Pull request by FridaTveit**
[Description](https://github.com/exercism/problem-specifications/blob/master/exercises/bank-account/description.md)

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by stkent** [comment]
[FYI: Kotlin has [an implementation of this exercise](https://github.com/exercism/kotlin/tree/master/exercises/bank-account) that might be inspirational, but I'm not sure if it's compliant with any canonical data!]

File: config.json [hunk 1]
```diff
@@ -374,6 +374,13 @@
       ]
     },
     {
+    	"slug": "bank-account",
+    	"difficulty": 6,
+    	"topics": [
+
+    	]
+    }
```
Comment (stkent) [inline config.json]: Missing comma here!

File: exercises/bank-account/src/test/java/BankAccountTest.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    @Test
+    public void newlyOpenedAccountHasEmptyBalance() throws BankAccountActionInvalidException {
+        bankAccount.open();
+
+        assertEquals(0, bankAccount.getBalance());
+    }
+
+    @Ignore
```
Comment (stkent) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: All `@Ignore` annotations should include the standard hint: `"Remove to run test"` :)

File: exercises/bank-account/src/test/java/BankAccountTest.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    @Test
+    public void canAdjustBalanceConcurrently() throws BankAccountActionInvalidException, InterruptedException {
+        bankAccount.open();
+        bankAccount.deposit(1000);
+
+        for (int i = 0; i < 10; i++) {
+            adjustBalanceConcurrently();
```
Comment (stkent) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: Does this test fail 100% of the time? If so, would you mind explaining how that works? I'm curious!

Comment (stkent) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: Ah, I see from the hint it is not guaranteed to fail. Can we provide any recommendation as to how many times a user should run this test to be "reasonably certain" their implementation is threadsafe? Alternatively, can we inject a little more thread control into solutions to be able to pull the appropriate levers ourselves and guarantee a concurrent modification attempt?

File: exercises/bank-account/src/test/java/BankAccountTest.java (hunk 2 above)
Comment (FridaTveit) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: Sorry, forgot! Will fix :)

File: exercises/bank-account/src/test/java/BankAccountTest.java (hunk 3 above)
Comment (FridaTveit) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: It failed every time I ran it (I think I ran it about 10 times in a row) with a non-threadsafe implementation. Is there any way we can be absolutely guarantee a concurrent modification attempt? I would expect the test to fail every time if their implementation isn't thread safe, but I can't be 100% certain. I could put in the hint that they should run the test a couple of times just to be safe?

Comment (stkent) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: Cool, I see the same behavior here. Let's add to the hint just to be safe :)

Comment (FridaTveit) [inline exercises/bank-account/src/test/java/BankAccountTest.java]: Will do :)

**Comment by stkent** [comment]
Love it! Great add.

**Pull request merged**